app/routes/CategoryRouter.ts

```diff
@@ -1,56 +1,87 @@
 'use strict';
-import { Router, Request, Response, NextFunction } from 'express';
+import { Request, Response, NextFunction } from 'express';
+import { Repository } from 'typeorm';
+import { Dependencies } from '../Types';
 import Category from '../orm/entities/Category';
-import { getRepository } from 'typeorm';
-import { validate } from 'class-validator';
+import Boom from 'boom';
 
-//Router to handle all category REST-api calls
 class CategoriesRouter {
-  constructor() {}
+  private categoryRepo: Repository<Category>;
+  constructor(opts: Dependencies) {
+    this.categoryRepo = opts.categoryRepo;
+  }
 
-  //Get all categories from the database
-  public async getAll(req: Request, res: Response, next: NextFunction) {
+  public getAll = async (req: Request, res: Response, next: NextFunction) => {
     try {
-      let results = await getRepository(Category).find();
+      let results = await this.categoryRepo.find();
       res.status(200).send(results);
     } catch (error) {
-      res.status(400).send({ message: "Couldn't get the data" });
+      const boom = Boom.boomify(error).output;
+      res.status(boom.statusCode).send(boom.payload);
```

Comment:
Same thing here that errormiddleware takes care of sending stuff back to client when error happens. That is it's job :) Fix all similar situations that are left